autoload/youcompleteme.vim

```diff
@@ -67,6 +67,7 @@ function! youcompleteme#Enable()
   endif
 
   call youcompleteme#EnableCursorMovedAutocommands()
+  call youcompleteme#EnableTextChangedAutocommands()
```

Comment:
@vheon I didn't get how this new state is different from the previous that could brake something that the former state coudn't. I've just added this new functions because their names fits better for what they're trying to achieve, but besides the naming and the spliting, it would just be doing the same it was all in the former functions, with their original names. I've checked where Enable/Disable functions are being used, and the new ones are in the same place, so I don't get how could anything be different it they were just one.